### PR TITLE
[IMP] Studio: add a data table using the report editor

### PR DESCRIPTION
**Pull request by auva-odoo**
task-3698568

### CODE REVIEW

File: content/applications/productivity/studio/pdf_reports.rst [hunk 1]
```diff
@@ -232,6 +232,36 @@ fields if needed. Then, specify the default value and press `Enter`.
...
+
+.. image:: pdf_reports/dynamic-table.png
+   :alt: List of available fields for the Tag model.
+
+.. note::
+   - The default row automatically iterates over the field's content, generating a row on the report
+     for each field value (e.g., one row per tag).
+   - Any row you add using the table control UI is added at the end of the table, but you can move
+     it using the table control UI.
+   - You can also data tables by :ref:`modifying the report's XML <studio/pdf-reports/XML-editing>`.
```
Comment (bouvyd) [inline content/applications/productivity/studio/pdf_reports.rst:263]:
```suggestion
   - You can also add data tables by :ref:`modifying the report's XML <studio/pdf-reports/XML-editing>`.
```

File: content/applications/productivity/studio/pdf_reports.rst [hunk 2]
```diff
@@ -232,6 +232,36 @@ fields if needed. Then, specify the default value and press `Enter`.
...
+<studio/pdf-reports/add-field>` in the columns. The dialog box that opens shows the source object
+for the field (e.g., the *Tag* model) and the list of available fields.
+
+.. image:: pdf_reports/dynamic-table.png
+   :alt: List of available fields for the Tag model.
+
+.. note::
+   - The default row automatically iterates over the field's content, generating a row on the report
+     for each field value (e.g., one row per tag).
+   - Any row you add using the table control UI is added at the end of the table, but you can move
```
Comment (bouvyd) [inline content/applications/productivity/studio/pdf_reports.rst:261]: not sure i'd mention this at all; the table tools give the option to 'insert above' or 'insert below' anyway - normally the tool should be self explanatory

File: content/applications/productivity/studio/pdf_reports.rst [hunk 3]
```diff
@@ -232,6 +232,36 @@ fields if needed. Then, specify the default value and press `Enter`.
...
+You can then insert the :ref:`field of your choice
+<studio/pdf-reports/add-field>` in the columns. The dialog box that opens shows the source object
+for the field (e.g., the *Tag* model) and the list of available fields.
+
+.. image:: pdf_reports/dynamic-table.png
+   :alt: List of available fields for the Tag model.
+
+.. note::
+   - The default row automatically iterates over the field's content, generating a row on the report
+     for each field value (e.g., one row per tag).
```
Comment (bouvyd) [inline content/applications/productivity/studio/pdf_reports.rst:260]: i'd remove the following point and extend this one a bit
```suggestion
     for each field value (e.g., one row per tag) - you can add static-content rows above or below the generated rows using the table tools
```

**Comment by auva-odoo** [comment]
@robodoo r+

**Pull request closed without merging**